helix-term/src/commands.rs

```diff
@@ -1179,11 +1202,19 @@ fn search(cx: &mut Context) {
     // HAXX: sadly we can't avoid allocating a single string for the whole buffer since we can't
     // feed chunks into the regex yet
     let contents = doc.text().slice(..).to_string();
+    let completions = search_completions(cx, Some(reg));
 
     let prompt = ui::regex_prompt(
         cx,
         "search:".into(),
         Some(reg),
+        move |input: &str| {
+            completions
+                .iter()
+                .filter(|comp| comp.starts_with(input))
+                .map(|comp| (0.., std::borrow::Cow::Owned(comp.clone())))
```

Comment:
Could we use `Cow::Borrowed` here?

Reply:
https://github.com/helix-editor/helix/blob/3eb829e2330fed5ad1c095f8bba44f62361b4943/helix-term/src/ui/prompt.rs#L15

prompt::Competion is required `'static`, so Lifetime here is shorter